cdk/stacks/current_elections.py

```diff
@@ -197,25 +246,75 @@ def __init__(self, scope: Construct, construct_id: str, **kwargs) -> None:
             timeout=Duration.minutes(10),
         )
 
-        one_am = aws_events.Schedule.cron(
-            minute="0", hour="1", timezone="Europe/London"
-        )
+        self.make_event_triggers()
+
+    @staticmethod
+    def glue_tables() -> List[GlueTable]:
+        return [current_ballots, current_ballots_joined_to_address_base]
+
+    @staticmethod
+    def s3_buckets() -> List[S3Bucket]:
+        return [ee_data_cache_production, data_baker_results_bucket]
+
+    def make_event_triggers(self):
+        one_am = aws_events.Schedule.cron(minute="0", hour="1")
         run_nightly_rule = aws_events.Rule(
             self, "RebuildCurrentElectionsNightlyTrigger", schedule=one_am
         )
 
         run_nightly_rule.add_target(
             aws_events_targets.SfnStateMachine(
-                state_machine=self.step_function,
+                machine=self.step_function,
                 # If we want to pass data to the step function, we can here:
                 # input=aws_events.RuleTargetInput.from_object({"trigger": "nightly"})
             )
         )
```

Comment:
This is a musing at this stage, rather than a request for change but..

One of the possible race conditions we're trying to avoid here is that we push an update on to the queue and also run on a schedule at the same time. It is quite unlikely that we will be editing EE at 1am, but lets imagine this was running hourly rather than daily.

At the way this works at the moment, we directly invoke a step function on a schedule.
Would it make more sense for the scheduled task to push an event onto the queue instead of invoking the function directly. Then we would benefit from SQS's built in de-duplication and sequential-ness. There can only be one message at the head of the queue.

If that makes things loads more complicated, maybe lets not do that right now, but something to think about..